framework/resource/resource.go

```diff
@@ -149,6 +152,42 @@ func (r *Resource) Destroy(args ...interface{}) error {
 	return result.Err()
 }
 
+// Status returns a copy of this resources' status, or nil if no status exists.
+func (r *Resource) Status() *pb.StatusReport_Resource {
```

Comment:
A resource's Status() func will need to be able to return multiple StatusReport_Resources. For example, a K8s deployment resource's Status() func will need to be able to return a Deployment status report resource and multiple Pod resources.

Reply:
Let's sync and discuss this next week, if you have time? The design was always to have 1:1 pair of `StatusReport_Resources{}` to resources defined in the resource manager using `WithResource{}`, which is what we have here.